tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -967,6 +1016,18 @@ static int handleTestInvocation(TestOptions Opts, TestOptions &InitOpts) {
     }
   }
 
+  if (!Opts.VFSFiles.empty()) {
+    sourcekitd_request_dictionary_set_string(Req, KeyVFSName, "testvfs");
+    sourcekitd_object_t VFSArgs = sourcekitd_request_array_create(nullptr, 0);
+    for (auto &NameAndTarget : Opts.VFSFiles) {
+      sourcekitd_request_array_set_string(VFSArgs, SOURCEKITD_ARRAY_APPEND,
```

Comment:
What do you think of using a dictionary instead of a list of strings to provide these options?  That seems like a better API rather than having to do option parsing from strings.  We use `OptionsDictionary` to represent an opaque settings dictionary elsewhere in sourcekit; maybe we could reuse that here?